worker-batch/src/main/java/com/hpe/caf/worker/batch/BatchWorker.java

```diff
@@ -15,53 +15,55 @@
  */
 package com.hpe.caf.worker.batch;
 
-import com.google.common.cache.LoadingCache;
 import com.hpe.caf.api.Codec;
 import com.hpe.caf.api.worker.*;
 import com.hpe.caf.worker.AbstractWorker;
-import com.rabbitmq.client.Connection;
 
 import java.util.Map;
 
 public class BatchWorker extends AbstractWorker<BatchWorkerTask, BatchWorkerResult> {
 
     private final BatchWorkerServicesImpl batchWorkerServices;
     private final Map<String, BatchWorkerPlugin> registeredPlugins;
-    private final TrackingInfo tracking;
-    private final BatchWorkerPublisher messagePublisher;
     private final BatchWorkerConfiguration configuration;
+    private String jobId;
```

Comment:
This needs to be `final`.

Reply:
That can't be final because it is being set in the catch block of the try, it gives an error because it may have already been assigned.

Reply:
Then create a private static `getJobId()` method that takes a `WorkerTaskData` or a `TrackingInfo` and returns the job id.  It needs to be made `final` to convey the semantics that it does not change after it has been initialized.